File: src/messages/global_state.js
import {GitHubState} from "./github_state.js";

export class GlobalState {

    constructor() {
        this._projects = new Map();
        this._github = new GitHubState();
    }

    get projects() {
        return this._projects;
    }

    set projects(value) {
        this._projects = value;
    }

    get github() {
        return this._github;
    }

    set github(value) {
        this._github = value;
    }
}

File: src-tauri/src/messages/global_state.rs
use crate::common::ProjectId;
use crate::messages::github_state::GitHubState;
use crate::messages::project_state::ProjectState;
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct GlobalState {
    pub github: GitHubState,
    pub projects: HashMap<ProjectId, ProjectState>,
}
